### ISSUE TITLE
Precision on string parsing

### ISSUE THREAD

**Issue by serpilliere** [bnjjj/twelf#29]
Hi @bnjjj,
First of all, thanks for your work on twelf!
I have a question regarding `string` arguments.
I have a simple config file defining a string argument:
```rust
#[derive(Parser, Debug)]
#[clap(author, version, about)]
pub struct ClientArgs {
...
    #[clap(
        long,
        short = 'j',
        help = r#"..."#
    )]
    pub allow_print: Option<String>,
...
```

I read the arguments (like in the example) from command line, environment variable, or config file:
```rust
    let config =
        ClientConfig::with_layers(
            &[
                Layer::Toml("./test.toml".into()),
                Layer::Env(Some(String::from("APP_"))),
                Layer::Clap(matches)
            ]
        )
        .unwrap();

```
The thing is:
- If I set allow_print from a toml file:
```toml
allow_print="test\\test"
```
when I run the program, I have:
```
  allow_print: Some("test\\test")
 ```
which show the back quote has been escaped so that at the end we have 1 backquote in the string (the output is debug here)

- If I set  allow_print from the environment variable:
```
APP_allow_print="test\testenv"
```
 I have:
```
allow_print: Some("test\\testenv")
```
which seems legit: We also have one backquote in the result (the output is still debug here)

But If I give argument from the command line:
```
--allow-print "test\testcmd"
```
I have:
```
allow_print: Some("test\\\\testcmd")
```
So here, the string from the argument seems to have been escaped a first time at parsing, to that the output here is double escaped.

I have double checked against clap, and it seems that clap doesn't escape string:
```
 --name "toto\titi"
```
gives, with this code:
```rust
use clap::Parser;

/// Simple program to greet a person
#[derive(Parser, Debug)]
#[command(author, version, about, long_about = None)]
struct Args {
    /// Name of the person to greet
    #[arg(short, long)]
    name: String,

    /// Number of times to greet
    #[arg(short, long, default_value_t = 1)]
    count: u8,
}

fn main() {
    let args = Args::parse();

    for _ in 0..args.count {
        println!("Hello {:?}!", args.name)
    }
}

```
the result:
```
Hello "toto\\titi"!
```

I am doing something wrong here or have I missed something?

Thanks again for your work!

**Comment by bnjjj** [comment]
@serpilliere thanks for detecting this issue. I just published a new version (`0.11.0`)  which fixes this double escape. Let me know if it works for you too.

**Comment by serpilliere** [comment]
Oh, thanks for the quick fix & version bumping @bnjjj !
I just tested it, it works here!
